examples/alpaca/alpaca.py

```diff
@@ -128,7 +128,15 @@ def _get_data_source(path_or_id):
     """The original alpaca.py used a json file, but it's since been moved to the HF dataset hub. You can use any
     dataset that's compatible with the structure of the alpaca dataset."""
     if fsspec_utils.exists(path_or_id):
-        return JsonDataset([path_or_id])
+        # get file format: jsonl or json
+        if path_or_id.endswith(".jsonl"):
```

Comment:
would probably be ideal to tweak TextUrlDataset or something to do the compression sniffing